pontoon/contributors/views.py

```diff
@@ -310,27 +311,38 @@ def settings(request):
     all_locales = list(Locale.objects.all())
     all_locales.insert(0, default_homepage_locale)
 
-    # Set custom homepage selector value
-    custom_homepage_locale = profile.custom_homepage
-    if custom_homepage_locale:
-        custom_homepage_locale = Locale.objects.filter(
-            code=custom_homepage_locale
-        ).first()
-    else:
+    # Set default for custom homepage locale based on code
+    try:
+        if profile.custom_homepage:
+            custom_homepage_locale = get_locale_or_redirect(profile.custom_homepage)
+        else:
+            custom_homepage_locale = default_homepage_locale
+    except Http404:
         custom_homepage_locale = default_homepage_locale
+        messages.info(
+            request,
+            "Your previously selected custom homepage locale is no longer available. Reverted to default.",
```

Comment:
We don't really revert the locale, we just show the default homepage when that happens. The user still needs to press Save. So I'd replace "Reverted to default." with something along the lines of "Please pick a different one.".